src/scales/scale.ts

```diff
@@ -200,4 +200,15 @@ export class Scale<D, R> {
     this._autoDomainIfAutomaticMode();
     return this;
   }
+
+  /**
+   * Whether Scale is comparable intuitively. eg. Quantitative vs Category, Time vs Time.
```

Comment:
@hellochar Scales represent an underlying data type (numbers, time, categories...) which can be comparable. There wasn't any reasoning in the original PR but I can imagine if the xAxis is linear and the yAxis logarithmic, then the "closest data point" is no longer just the pixel count.

There wasn't a unit test for this, but I can write one with a full blown chart once we decide that this is the right format / functionality we're going for.

On further thought, I think we should actually just introduce a flag in `entityNearest` on whether to use data space or pixel space. There's no way we can tell if they're comparable without the user data, and we should just allow the option for both. Thoughts?